OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -1926,6 +1928,49 @@ void testAliasesAndLabels() {
     SimTK_TEST(foo->getInput("listInput1").getLabel(1) == "thud");
 }
 
+void testSignalGenerator() {
+    // Feed a SignalGenerator's output into a TableReporter, and make sure the
+    // reported value is correct.
+
+    // Build the model.
+    TheWorld world;
+    auto* signalGen = new SignalGenerator();
+    signalGen->setName("sinusoid");
+    const double amplitude = 1.5;
+    const double omega = 3.1;
+    const double phase = 0.3;
+    signalGen->set_function(Sine(amplitude, omega, phase));
+    world.addComponent(signalGen);
+    
+    auto* reporter = new TableReporter();
+    reporter->addToReport(signalGen->getOutput("signal"));
+    world.addComponent(reporter);
+
+    // Build the system.
+    // Check that the value of the input is the same as before.
```

Comment:
There's no check (e.g., `SimTK_TEST_EQ()`) in this block and the system is being realized to only `Stage::Model` so it isn't clear (to me, at least) how L1951–1954 contribute to checking the value of the input. Perhaps remove this comment?